com.unity.render-pipelines.universal/ShaderLibrary/Core.hlsl

```diff
@@ -103,6 +103,77 @@
     #define GATHER_BLUE_TEXTURE2D_X(textureName, samplerName, coord2)       GATHER_BLUE_TEXTURE2D(textureName, samplerName, coord2)
 #endif
 
+///
+/// Texture Sampling Macro Overrides for Scaling
+///
+/// When mip bias is supported by the underlying platform, the following section redefines all 2d texturing operations to support a global mip bias feature.
+/// This feature is used to improve rendering quality when image scaling is active. It achieves this by adding a bias value to the standard mip lod calculation
+/// which allows us to select the mip level based on the final image resolution rather than the current rendering resolution.
+
+#ifdef PLATFORM_SAMPLE_TEXTURE2D_BIAS
```

Comment:
I wonder why we can't move these two Core RP. Both pipelines support _GlobalMipBias and _GlobalMipBiasPow2 right?

Reply:
I did investigate moving this into core a bit, but ultimately we have two choices that I could think of. We can slip it into Common.hlsl and let it inject itself into basically all shaders and gate the override by a define like HDRP does, or we could add a new core hlsl file that's optionally included and HDRP and URP can include it in different spots but it would be opt-in. The problem I see with both of these is the fact that the setup of the value itself from the CPU would be totally different from either SRP variant and if we did the auto-enable version, custom SRPs could forget to set the value entirely.

I think the ideal solution in the long term is probably to revisit our approach for mip bias and to find a solution that isolates material texel fetches rather than using a global override like this. The current implementation affects sampling in post process shaders and any other shaders as well since it's just overriding the main sampling macros. If we could find a way to isolate the bias logic to just material fetches with a new macro like "SAMPLE_MATERIAL_2D(x)", I think it'd make things a lot safer overall. Maybe we could also include the CPU-side constant setup inside Core as part of the dynamic resolution scaling logic?

Reply:
Yes, that makes sense if we plan to revisit this logic then it's best to do it before moving to core. 

> Maybe we could also include the CPU-side constant setup inside Core as part of the dynamic resolution scaling logic?

Yes this makes sense to me. @sandy-carter-unity for awareness.